pycbc/conversions.py

```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
     ss[mask] = spin1[mask]
     return _formatreturn(ss)
 
+def primary_xi(spinx, spiny):
+    """Returns the effective precession spin argument for the larger mass.
+    """
+    return chi_perp_from_spin1x_spin1y(spinx, spiny)
+
+def secondary_xi(mass1, mass2, spinx, spiny):
+    """Returns the effective precession spin argument for the smaller mass.
+    """
+    q = q_from_mass1_mass2(mass1, mass2)
+    a1 = 2 + 3 * q / 2
+    a2 = 2 + 3 / (2 * q)
+    return a1 / (q**2 * a2) * chi_perp_from_spin1x_spin1y(spinx, spiny)
+
+def chi_perp_from_spin1x_spin1y(spin1x, spin1y):
+    """Returns the in-plane spin from spin1x and spin1y.
+    """
+    return numpy.sqrt(spin1x**2 + spin1y**2)
+
+def chi_perp_from_spin2x_spin2y(spin2x, spin2y):
+    """Returns the in-plane spin from spin2x and spin2y.
+    """
+    return chi_perp_from_spin1x_spin1y(spin2x, spin2y)
```

Comment:
These two functions are redundant. Rename `chi_perp_from_spin1x_spin1y`, `chi_perp_from_spinx_spiny`, and get rid of the second function.